### PR TITLE
Added an alias for indexing by groups

### PR DESCRIPTION
**Pull request by sagittaracc**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️/❌
| New feature?  | ✔️/❌
| Breaks BC?    | ✔️/❌
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

### CODE REVIEW

File: src/ArrayHelper.php [hunk 1]
```diff
@@ -713,6 +713,21 @@ public static function index(array $array, $key, $groups = []): array
...
     }
 
+    /**
+     * Groups the array according to a specified key.
+     * This is just an alias for indexing by groups
+     *
+     * @param array $array The array that needs to be grouped.
+     * @param Closure[]|string|string[] $groups The array of keys, that will be used to group the input array
+     * by one or more keys.
+     *
```
Comment (vjik) [inline src/ArrayHelper.php:723]:
```suggestion
     *
     * @psalm-param array<mixed, array|object> $array
     *
```

**Review by vjik — CHANGES_REQUESTED** [review]
Can you add tests for this method?

File: src/ArrayHelper.php (hunk 1 above)
Comment (sagittaracc) [reply src/ArrayHelper.php:723]: I did

**Review by vjik — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thanks.

**Pull request merged**